elixir/apps/api/lib/api/client/channel.ex

```diff
@@ -473,6 +414,110 @@ defmodule API.Client.Channel do
     end
   end
 
+  #############################################################
+  ##### Forwarding replies from the gateway to the client #####
+  #############################################################
+
+  # This the list of ICE candidates gathered by the gateway and relayed to the client
+  def handle_info(
+        {:ice_candidates, gateway_id, candidates, {opentelemetry_ctx, opentelemetry_span_ctx}},
+        socket
+      ) do
+    OpenTelemetry.Ctx.attach(opentelemetry_ctx)
+    OpenTelemetry.Tracer.set_current_span(opentelemetry_span_ctx)
+
+    OpenTelemetry.Tracer.with_span "client.ice_candidates",
+      attributes: %{
+        gateway_id: gateway_id,
+        candidates_length: length(candidates)
+      } do
+      push(socket, "ice_candidates", %{
+        gateway_id: gateway_id,
+        candidates: candidates
+      })
+
+      {:noreply, socket}
+    end
+  end
+
+  def handle_info(
+        {:invalidate_ice_candidates, gateway_id, candidates,
+         {opentelemetry_ctx, opentelemetry_span_ctx}},
+        socket
+      ) do
+    OpenTelemetry.Ctx.attach(opentelemetry_ctx)
+    OpenTelemetry.Tracer.set_current_span(opentelemetry_span_ctx)
+
+    OpenTelemetry.Tracer.with_span "client.invalidate_ice_candidates",
+      attributes: %{
+        gateway_id: gateway_id,
+        candidates_length: length(candidates)
+      } do
+      push(socket, "invalidate_ice_candidates", %{
+        gateway_id: gateway_id,
+        candidates: candidates
+      })
+
+      {:noreply, socket}
+    end
+  end
+
+  # DEPRECATED IN 1.4
+  # This message is sent by the gateway when it is ready to accept the connection from the client
+  def handle_info(
+        {:connect, socket_ref, resource_id, gateway_public_key, payload,
+         {opentelemetry_ctx, opentelemetry_span_ctx}},
+        socket
+      ) do
+    OpenTelemetry.Ctx.attach(opentelemetry_ctx)
+    OpenTelemetry.Tracer.set_current_span(opentelemetry_span_ctx)
+
+    OpenTelemetry.Tracer.with_span "client.connect", attributes: %{resource_id: resource_id} do
+      reply(
+        socket_ref,
+        {:ok,
+         %{
+           resource_id: resource_id,
+           persistent_keepalive: 25,
+           gateway_public_key: gateway_public_key,
+           gateway_payload: payload
+         }}
+      )
+
+      {:noreply, socket}
+    end
+  end
+
+  def handle_info(
+        {:connect, socket_ref, resource_id, gateway_id, gateway_public_key, preshared_key,
+         ice_credentials, {opentelemetry_ctx, opentelemetry_span_ctx}},
+        socket
+      ) do
+    OpenTelemetry.Ctx.attach(opentelemetry_ctx)
+    OpenTelemetry.Tracer.set_current_span(opentelemetry_span_ctx)
+
+    OpenTelemetry.Tracer.with_span "client.connect", attributes: %{resource_id: resource_id} do
+      reply(
```

Comment:
Instead of a reply, would it be very difficult to just send a new message? The resource ID is enough context for us and new messages are easier to deserialise than replies.